lib/Sema/CSSimplify.cpp

```diff
@@ -6034,36 +6052,137 @@ ConstraintSystem::simplifyCheckedCastConstraint(
       break;
   } while (true);
 
+  auto attemptFixForCheckedConvertibleTypes = [&]() -> ConstraintFix * {
```

Comment:
I suggest we make this a static function instead of avoid polluting body of `simplifyCheckedCastConstraint` with diagnostic logic.

Reply:
Also fixed